### DISCUSSION TITLE
VFX and particle system?

### DISCUSSION THREAD

**Discussion by watchmax3** [o3de/o3de#13960] [Help Wanted] [answered]
1- popcornfx plugin let free commercial use or not? is still Supported?                                                                                                      2- openfx from The Academy Software Foundation hosts projects is just for movie or also is for game engines?                                                 3- i have idea: spawn very tiny objects all over level and use nvdia physX for wind simulation but i think create heavy render. is nice idea or not?                                                                                                                                                                                                     4- there was a particle system in lumberyard now is r.i.p. why still is not a particle system and vfx system in o3de?! u can easy see create fire or smoke in unreal Tutorials in internet ....but here?

**Comment by Darkinggq** [comment]
>  4- there was a particle system in lumberyard now is r.i.p. why still is not a particle system and vfx system in o3de?! u can easy see create fire or smoke in unreal Tutorials in internet ....but here?

they say there will be native particles, maybe not soon  https://github.com/o3de/o3de/issues/2889

i don't know why the lumberyard particle system is not porting to o3de they are gone or removed due to new Atom render (have to rewrite new particle system to o3de from scratch)

**Comment by amzn-tommy** [comment, marked as answer]
1. Since there is no native solution, PopcornFX is still the recommended tool for particle effects since they integrate with o3de https://www.popcornfx.com/plugin-o3de/. If you have questions about their license, I would recommend reaching out to PopcornFX directly.

2. ASWF is focused on motion pictures, though there is some overlap and long term potential to support some of those technologies in real-time applications. There is a motion picture industry vertical channel on the O3DE Discord server, which is another good place to discuss that overlap https://discord.com/channels/805939474655346758/821693440462815243

3. Like most things, that depends on your content and desired result, such as how many objects you need, the complexity of the material used to render those objects, the number of triangles, number of objects, performance of the wind simulation, whether or not there are other physics bodies you need the objects to interact with, how much of the simulation could be baked vs. needing to update interactively, etc.

O3DE does not have support for instancing with the mesh component (though that is a work in progress), so it would result in a lot of draw calls. There is also overhead in activating mesh and material components that would need to be optimized. So it would indeed result in heavy overhead right now. We're working to improve that in the context of things like the vegetation system spawning in lots of instances of the same object, but it may not be enough for your use-case.

4. The particle system from Lumberyard would have to be re-written from scratch to work with Atom. You could potentially try to keep the original tooling intact, but that was generally considered one of the least enjoyable aspects of the particle system compared to a node based editor like PopcornFX. Either way, that would be a large effort, which is one of the reasons why no one has done that.
